tests/e2e/performance/test_pvc_bulk_clone_performance.py

```diff
@@ -0,0 +1,196 @@
+    """
+    Base class for bulk creation of PVC clones
+    """
+
+    @pytest.fixture()
+    def namespace(self, project_factory, interface_iterate):
+        """
+        Create a new project
+        """
+        proj_obj = project_factory()
+        self.namespace = proj_obj.namespace
+        self.interface = interface_iterate
+
+    @pytest.mark.usefixtures(namespace.__name__)
+    def test_bulk_clone_perforamnce(self, namespace, tmp_path, pod_factory):
+        """
+        Creates number of PVCs in a bulk using kube job
+        Write 60% of PVC capacity to each one of the created PVCs
+        Creates 1 clone per each PVC altogether in a bulk
+        Measuring time for bulk of clones creation
+
+        """
+        pvc_count = 50
+        log.info(f"Start creating {self.interface} {pvc_count} PVC")
+        if self.interface == constants.CEPHBLOCKPOOL:
+            sc_name = constants.DEFAULT_STORAGECLASS_RBD
+            clone_yaml = constants.CSI_RBD_PVC_CLONE_YAML
+        elif self.interface == constants.CEPHFILESYSTEM:
+            sc_name = constants.DEFAULT_STORAGECLASS_CEPHFS
+            clone_yaml = constants.CSI_CEPHFS_PVC_CLONE_YAML
+
+        pvc_dict_list = scale_lib.construct_pvc_creation_yaml_bulk_for_kube_job(
+            no_of_pvc=pvc_count,
+            access_mode=constants.ACCESS_MODE_RWO,
+            sc_name=sc_name,
+            pvc_size="5Gi",
+        )
+
+        job_pvc_file = ObjectConfFile(
+            name="job_profile_pvc",
+            obj_dict_list=pvc_dict_list,
+            project=self.namespace,
+            tmp_path=tmp_path,
+        )
+
+        # Create kube_job
+        job_pvc_file.create(namespace=self.namespace)
+
+        # Check all the PVC reached Bound state
+        pvc_bound_list = scale_lib.check_all_pvc_reached_bound_state_in_kube_job(
+            kube_job_obj=job_pvc_file,
+            namespace=self.namespace,
+            no_of_pvc=pvc_count,
+        )
+
+        logging.info(f"Number of PVCs in Bound state {len(pvc_bound_list)}")
+
+        total_files_size = self.run_fio_on_pvcs(pvc_dict_list, pod_factory)
+
+        clone_dict_list = self.construct_pvc_clone_yaml_bulk_for_kube_job(
+            pvc_dict_list, clone_yaml, sc_name
+        )
+
+        logging.info("Created clone dict list")
+
+        job_clone_file = ObjectConfFile(
+            name="job_profile_clone",
+            obj_dict_list=clone_dict_list,
+            project=self.namespace,
+            tmp_path=tmp_path,
+        )
+
+        # Create kube_job that creates clones
+        job_clone_file.create(namespace=self.namespace)
+
+        logging.info("Going to check bound status for clones")
+        # Check all the clones reached Bound state
+        clone_bound_list = scale_lib.check_all_pvc_reached_bound_state_in_kube_job(
+            kube_job_obj=job_clone_file,
+            namespace=self.namespace,
+            no_of_pvc=pvc_count,
+            timeout=600,
```

Comment:
600sec is huge timeout, In function we have 10 iterations for checking PVCs reached bound state or not, with that total timeout for the testcase to fail is 600 * 10 = 6000sec(i.e. 10hrs). waiting for 10hrs, we need to reduce this.

Reply:
@ramkiperiy - initially I was using this timeout for random IO size. 
Now when the test creates 5GB pvcs and write around 3GB data, I'll try to lower the timeout to 90 ( 900 sec = 15 mins instead of iriginal 5 mins for 1GB PVC) . 
If it passes, I'll leave it. If there sould be problems with FIO - I'll increase it.

Reply:
@ramkiperiy 
For CephPS this timeout is not enought  ( 50 PVCs have to reach bound state, in total 150GB of data needs to be written upon clone creation) : 
 Listed PVCs took more than 900 secs to bound . 

I'm increasing the timeout from 90 to 150 ( 150*10 = 1500secx = 25 mins)

Reply:
@ramkiperiy - 1500 secs is not enough. Not all clones reached BOUND state. 
Increasing the timeout to 50 mins ( 300 * 10)  and will try to run.

Reply:
@ramkiperiy - 50 mins is not enough for 50 CephFS clones with data:

 Listed PVCs took more than 3000 secs to bound 

Increasing the timeout to 400 ( 400*10=4000 secs = 1 hour 6 mins.

Reply:
- since the 4000 secs timeout was not enough for bounding 50 clones ( 5GB) each with data ( around 150GB data) , I'm increasing the timeout to 500 ( 500*10= 5000 secs).

Reply:
14:42:07 - MainThread - root - ERROR -  Listed PVCs took more than 5000 secs to bound

Increasing the timeout to 600.

Reply:
@ramkiperiy - let's currently have the timeout equal to 600. 
I'm going to test it on various platforms and will make it lower when I know exactlly the limit.

Reply:
@ramkiperiy  - eventually I've set timout to 200 since on Azure the clone bulk creation time is up to 2000 secs. 
If the test will fail on other platforms, we'll consider raising it.